### DISCUSSION TITLE
QSelect: hide disabled items from list

### DISCUSSION THREAD

**Discussion by Razi91** [quasarframework/quasar#15501] [Ideas / Proposals]
The situation is that I need to disable some items from being selected, but I need them in `options` to let them be displayed if they are already selected. I use `id` for `modelValue` and `map-options` prop (along with `option-value` and `option-label`) to display the proper label.

I would rather avoid writing my own slots for that.

It could be done either by:
1. `hide-disabled` prop, which would just not display them
2. `option-hidden` prop, which works similarly to `option-disabled`, but defines if the option should be hidden
3. provide `options-all` with a 'database' to use for such objects and `options` for options to display (no breaking changes, just additional prop to provide all data)

//EDIT:
And yes, I checked, the 1 and 2 might be problematic due to `virtualScrollSliceRange` being calculated for the wrong item count, `virtualScrollLength` would have to filter out the disabled options first, the `props.options` would have to be calculated in this case.

WDYT?
//EDIT2:
example of third suggestion:
https://codepen.io/razi91-the-selector/pen/RwYKega?editors=1010

**Comment by pdanpdan** [comment]
make a codepen please

**Reply by Razi91** [reply]
added: https://codepen.io/razi91-the-selector/pen/RwYKega?editors=1010

**Comment by pdanpdan** [comment]
You can:
- hide the disabled options with css (not very nice)
- use object values and a mapper (best way)

https://codepen.io/pdanpdan/pen/poORmwL?editors=1010

**Reply by Razi91** [reply]
The second solution isn't bad, the first one will break virtualscroll.

I think I'll stay with the first solution, at least for now, thanks
